app/commands/assets/soft_delete_relations.rb

```diff
@@ -0,0 +1,17 @@
+module Assets
+  class SoftDeleteRelations
+    attr_reader :asset
+
+    def initialize(asset:)
+      @asset = asset
+    end
+
+    def call
+      asset.comments&.map(&:soft_delete)
+      # first update playlists
+      asset.playlists.update_all(['tracks_count = tracks_count - 1, playlists.updated_at = ?', Time.now]) unless asset.playlists.empty?
+      asset.tracks&.map(&:soft_delete)
+      asset.listens&.map(&:soft_delete)
```

Comment:
`&.map(&:soft_delete)` will issue 1 sql call per relation, which is unideal, even if in a job vs. request cycle. We either need to devise a collection method in `soft_deletion.rb` or manually call `.update_all(deleted_at: time)`.

Reply:
Hmmm I guess my intention here was to be able to trigger any associated actions for `listens` or `tracks`. But since these things don't seem to have associations we might want to delete, then I think it's safe to just `update_all()`